### PR TITLE
Added changelog for partner SDK

### PR DESCRIPTION
**Pull request by fivetran-abdulsalam**
Links https://fivetran.height.app/T-708746
Add changelog file and June release

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#### June 2024
+
+-----------
+**ConfigurationForm Changes**
+- We added options to provide default value and placeholder for the Field.
+- We Introduced  FieldSet  support, which allows to group the Fields and add visibility condition to control displaying on the form.
+- The FormField format is updated to support FieldSet. The single attribute should be used to assign the usual fields.
```
Comment (manjutapali) [inline CHANGELOG.md:7]: NIT: can be changed from `usual fields` to `regular fields`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+**SourceConnector Changes**
+- We removed LogEntry from Update operation. Now, partners can use STD::OUT to send logs. For more details, please refer to this.
+- We renamed few enums and services:
```
Comment (manjutapali) [inline CHANGELOG.md:11]: We have renamed the few enums and services

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  - Renamed the service definition from Connector to SourceConnector.
+  - Renamed the OpType field to RecordType.
+- Added support for NAIVE_TIME datatype. Partners can now send data with this datatype.
+
+**DestinationConnector Changes**
+- Added NAIVE_TIME data type support.
+- We renamed the service definition from Destination to DestinationConnector.
```
Comment (manjutapali) [inline CHANGELOG.md:18]: We have renamed the service

**Review by manjutapali — COMMENTED** [review]
added comments

File: CHANGELOG.md (hunk 1 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: incorporated

File: CHANGELOG.md (hunk 3 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: incorporated

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#### June 2024
+
+-----------
+**ConfigurationForm Changes**
+- We added options to provide default value and placeholder for the Field.
+- We Introduced FieldSet support, which allows to group the Fields and add visibility condition to control displaying on the form.
+- The FormField format is updated to support FieldSet. The single attribute should be used to assign the regular fields.
+
+**SourceConnector Changes**
+- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to [this](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging).
```
Comment (ediril) [inline CHANGELOG.md:10]:
```suggestion
- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+- We have renamed the few enums and services:
+  - Renamed the service definition from `Connector` to `SourceConnector`.
+  - Renamed the `OpType` field to `RecordType`.
+- Added support for `NAIVE_TIME` datatype. Partners can now send data with this datatype.
+
+**DestinationConnector Changes**
+- Added `NAIVE_TIME` data type support.
```
Comment (ediril) [inline CHANGELOG.md:17]:
```suggestion
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+- We added options to provide default value and placeholder for the Field.
+- We Introduced FieldSet support, which allows to group the Fields and add visibility condition to control displaying on the form.
+- The FormField format is updated to support FieldSet. The single attribute should be used to assign the regular fields.
+
+**SourceConnector Changes**
+- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to [this](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging).
+- We have renamed the few enums and services:
+  - Renamed the service definition from `Connector` to `SourceConnector`.
+  - Renamed the `OpType` field to `RecordType`.
+- Added support for `NAIVE_TIME` datatype. Partners can now send data with this datatype.
```
Comment (ediril) [inline CHANGELOG.md:14]:
```suggestion
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+-----------
+**ConfigurationForm Changes**
+- We added options to provide default value and placeholder for the Field.
+- We Introduced FieldSet support, which allows to group the Fields and add visibility condition to control displaying on the form.
+- The FormField format is updated to support FieldSet. The single attribute should be used to assign the regular fields.
+
+**SourceConnector Changes**
+- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to [this](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging).
+- We have renamed the few enums and services:
```
Comment (ediril) [inline CHANGELOG.md:11]:
```suggestion
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  - Renamed the service definition from `Connector` to `SourceConnector`.
```
Comment (ediril) [inline CHANGELOG.md:12]:
```suggestion
- Renamed service definition from `Connector` to `SourceConnector`.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  - Renamed the `OpType` field to `RecordType`.
```
Comment (ediril) [inline CHANGELOG.md:13]:
```suggestion
- Renamed `OpType` to `RecordType`.
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+- Added support for `NAIVE_TIME` datatype. Partners can now send data with this datatype.
+
+**DestinationConnector Changes**
+- Added `NAIVE_TIME` data type support.
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
+- A new rpc method `Capabilities` is introduced. This brings the functionalities:
```
Comment (ediril) [inline CHANGELOG.md:19]:
```suggestion
- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    - Adjust column types: Supports adjusting Partner data type for each Fivetran type.
```
Comment (ediril) [inline CHANGELOG.md:20]:
```suggestion
    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
+#### June 2024
+
+-----------
```
Comment (ediril) [inline CHANGELOG.md:3]:
```suggestion
-----------
+ - Added a new `NAIVE_TIME` data type.

```

**Review by ediril — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 12 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md:3]: Incorporated

File: CHANGELOG.md (hunk 4 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: Incorporated

File: CHANGELOG.md (hunk 7 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: Incorporated

File: CHANGELOG.md (hunk 8 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: Incorporated

File: CHANGELOG.md (hunk 9 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: Incorporated

File: CHANGELOG.md (hunk 6 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: Incorporated

File: CHANGELOG.md (hunk 5 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: Incorporated

File: CHANGELOG.md (hunk 11 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: Incorporated

File: CHANGELOG.md (hunk 10 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: Incorporated

File: CHANGELOG.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+**DestinationConnector Changes**
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
+- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the max value supported for the columns types.
+    - We have added support for providing parquet batch files. Now, partner can choose the file type to create batch files.
+    - Introduced a new field `supports_history_mode`, which indicates whether the destination supports history mode.
+- Changes in AlterTableRequest:
+    - Takes `table_name` instead of `Table`
+    - Introduced SchemaDiff field, which supports adding new columns, change column type, and update primary keys.
```
Comment (5tran-alexil) [inline CHANGELOG.md:25]:
```suggestion
    - Introduced a new field, `SchemaDiff`, which supports adding new columns, change column type, and update primary keys.  
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -0,0 +1,28 @@
+#### June 2024
+
+-----------
+- Added a new `NAIVE_TIME` data type.
+
+**ConfigurationForm Changes**
+- We added options to provide default value and placeholder for the Field.
+- We Introduced FieldSet support, which allows to group the Fields and add visibility condition to control displaying on the form.
```
Comment (5tran-alexil) [inline CHANGELOG.md:8]:
```suggestion
- We introduced FieldSet support, which allows to group the Fields and add visibility condition to control displaying on the form.
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+- Renamed service definition from `Connector` to `SourceConnector`.
+- Renamed `OpType` to `RecordType`.
+
+**DestinationConnector Changes**
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
+- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the max value supported for the columns types.
+    - We have added support for providing parquet batch files. Now, partner can choose the file type to create batch files.
+    - Introduced a new field `supports_history_mode`, which indicates whether the destination supports history mode.
```
Comment (5tran-alexil) [inline CHANGELOG.md:22]:
```suggestion
    - Introduced a new field, `supports_history_mode`, which indicates whether the destination supports history mode.
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+**SourceConnector Changes**
+- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
+- Renamed service definition from `Connector` to `SourceConnector`.
+- Renamed `OpType` to `RecordType`.
+
+**DestinationConnector Changes**
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
+- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the max value supported for the columns types.
```
Comment (5tran-alexil) [inline CHANGELOG.md:20]:
```suggestion
    - Define the maximal value supported for the column types.
```

File: CHANGELOG.md [hunk 17]
```diff
@@ -0,0 +1,28 @@
+#### June 2024
+
+-----------
+- Added a new `NAIVE_TIME` data type.
+
+**ConfigurationForm Changes**
+- We added options to provide default value and placeholder for the Field.
```
Comment (5tran-alexil) [inline CHANGELOG.md:7]:
```suggestion
- We added options to provide default value and placeholder for `Field`.
```

File: CHANGELOG.md [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+- We Introduced FieldSet support, which allows to group the Fields and add visibility condition to control displaying on the form.
+- The FormField format is updated to support FieldSet. The single attribute should be used to assign the regular fields.
```
Comment (5tran-alexil) [inline CHANGELOG.md:9]:
```suggestion
- The `FormField` format is updated to support `FieldSet`. A single attribute should be used to assign the regular fields.
```

File: CHANGELOG.md [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the max value supported for the columns types.
+    - We have added support for providing parquet batch files. Now, partner can choose the file type to create batch files.
+    - Introduced a new field `supports_history_mode`, which indicates whether the destination supports history mode.
+- Changes in AlterTableRequest:
+    - Takes `table_name` instead of `Table`
+    - Introduced SchemaDiff field, which supports adding new columns, change column type, and update primary keys.  
+- Changes in WriteBatchRequest:
+  - We have added support for parquet batch files configuration details.
+  - Introduced new field `HistoryMode`, which indicates history mode is enabled for table.
```
Comment (5tran-alexil) [inline CHANGELOG.md:28]:
```suggestion
  - We have introduced new field `HistoryMode`, which indicates that history mode is enabled for the table.
```

File: CHANGELOG.md [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+
+**DestinationConnector Changes**
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
+- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the max value supported for the columns types.
+    - We have added support for providing parquet batch files. Now, partner can choose the file type to create batch files.
+    - Introduced a new field `supports_history_mode`, which indicates whether the destination supports history mode.
+- Changes in AlterTableRequest:
+    - Takes `table_name` instead of `Table`
```
Comment (5tran-alexil) [inline CHANGELOG.md:24]:
```suggestion
    - It now uses `table_name` instead of `Table`.
```

File: CHANGELOG.md [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+- Renamed `OpType` to `RecordType`.
+
+**DestinationConnector Changes**
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
+- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the max value supported for the columns types.
+    - We have added support for providing parquet batch files. Now, partner can choose the file type to create batch files.
+    - Introduced a new field `supports_history_mode`, which indicates whether the destination supports history mode.
+- Changes in AlterTableRequest:
```
Comment (5tran-alexil) [inline CHANGELOG.md:23]:
```suggestion
- Changes in `AlterTableRequest`:
```

File: CHANGELOG.md [hunk 22]
```diff
@@ -0,0 +1,28 @@
...
+
+**ConfigurationForm Changes**
+- We added options to provide default value and placeholder for the Field.
+- We Introduced FieldSet support, which allows to group the Fields and add visibility condition to control displaying on the form.
+- The FormField format is updated to support FieldSet. The single attribute should be used to assign the regular fields.
+
+**SourceConnector Changes**
+- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
+- Renamed service definition from `Connector` to `SourceConnector`.
+- Renamed `OpType` to `RecordType`.
```
Comment (5tran-alexil) [inline CHANGELOG.md:14]:
```suggestion
- We have renamed `OpType` to `RecordType`.
```

File: CHANGELOG.md [hunk 23]
```diff
@@ -0,0 +1,28 @@
...
+-----------
+- Added a new `NAIVE_TIME` data type.
+
+**ConfigurationForm Changes**
+- We added options to provide default value and placeholder for the Field.
+- We Introduced FieldSet support, which allows to group the Fields and add visibility condition to control displaying on the form.
+- The FormField format is updated to support FieldSet. The single attribute should be used to assign the regular fields.
+
+**SourceConnector Changes**
+- We removed `LogEntry` from Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
```
Comment (5tran-alexil) [inline CHANGELOG.md:12]:
```suggestion
- We have removed `LogEntry` from the Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
```

**Comment by fivetran-abdulsalam** [comment]
Incorporated all the comments

File: CHANGELOG.md [hunk 24]
```diff
@@ -0,0 +1,28 @@
+#### June 2024
+
+-----------
+- Added a new `NAIVE_TIME` data type.
+
+**ConfigurationForm Changes**
+- We have added options to provide default value and placeholder for `Field`.
+- We have introduced `FieldSet` support, which allows to group the `Fields` and add visibility condition to control displaying on the form.
```
Comment (ediril) [inline CHANGELOG.md:8]:
```suggestion
- Introduced `FieldSet` support in `FormField`, which allows to group the `Fields` and add visibility condition to control displaying on the form.
```

File: CHANGELOG.md [hunk 25]
```diff
@@ -0,0 +1,28 @@
...
+- The `FormField` format is updated to support `FieldSet`. A single attribute should be used to assign the regular fields.
```
Comment (ediril) [inline CHANGELOG.md:9]:
```suggestion
```

File: CHANGELOG.md [hunk 26]
```diff
@@ -0,0 +1,28 @@
+#### June 2024
+
+-----------
+- Added a new `NAIVE_TIME` data type.
+
+**ConfigurationForm Changes**
+- We have added options to provide default value and placeholder for `Field`.
```
Comment (ediril) [inline CHANGELOG.md:7]:
```suggestion
- Added options to provide default value and placeholder for `Field`.
```

File: CHANGELOG.md [hunk 27]
```diff
@@ -0,0 +1,28 @@
...
+- We have introduced `FieldSet` support, which allows to group the `Fields` and add visibility condition to control displaying on the form.
+- The `FormField` format is updated to support `FieldSet`. A single attribute should be used to assign the regular fields.
+
+**SourceConnector Changes**
+- We have removed `LogEntry` from the Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
```
Comment (ediril) [inline CHANGELOG.md:12]:
```suggestion
- Removed `LogEntry` from the Update operation. Now, partners can use `STD::OUT` to send logs. For more details, please refer to the [logging](https://github.com/fivetran/fivetran_sdk/blob/main/development-guide.md#logging) section in the SDK Development Guide.
```

File: CHANGELOG.md [hunk 28]
```diff
@@ -0,0 +1,28 @@
...
+- We have renamed service definition from `Connector` to `SourceConnector`.
```
Comment (ediril) [inline CHANGELOG.md:13]:
```suggestion
- Renamed service definition from `Connector` to `SourceConnector`.
```

File: CHANGELOG.md [hunk 29]
```diff
@@ -0,0 +1,28 @@
...
+- We have renamed `OpType` to `RecordType`.
```
Comment (ediril) [inline CHANGELOG.md:14]:
```suggestion
- Renamed `OpType` to `RecordType`.
```

File: CHANGELOG.md [hunk 30]
```diff
@@ -0,0 +1,28 @@
...
+
+**DestinationConnector Changes**
+- We have renamed the service definition from `Destination` to `DestinationConnector`.
```
Comment (ediril) [inline CHANGELOG.md:17]:
```suggestion
- Renamed the service definition from `Destination` to `DestinationConnector`.
```

File: CHANGELOG.md [hunk 31]
```diff
@@ -0,0 +1,28 @@
...
+- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
```
Comment (ediril) [inline CHANGELOG.md:18]:
```suggestion
- Introduced a new rpc method `Capabilities` which brings the following functionalities:
```

File: CHANGELOG.md [hunk 32]
```diff
@@ -0,0 +1,28 @@
...
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the maximal value supported for the column types.
```
Comment (ediril) [inline CHANGELOG.md:20]:
```suggestion
    - Define maximum value supported by the destination column types.
```

File: CHANGELOG.md [hunk 33]
```diff
@@ -0,0 +1,28 @@
...
+    - We have added support for providing parquet batch files. Now, partner can choose the file type to create batch files.
```
Comment (ediril) [inline CHANGELOG.md:21]:
```suggestion
    - Added support for parquet batch files.
```

File: CHANGELOG.md [hunk 34]
```diff
@@ -0,0 +1,28 @@
...
+    - We have introduced a new field, `supports_history_mode`, which indicates whether the destination supports history mode.
```
Comment (ediril) [inline CHANGELOG.md:22]:
```suggestion
    - Introduced a new field, `supports_history_mode`, which indicates whether the destination supports history mode.
```

File: CHANGELOG.md [hunk 35]
```diff
@@ -0,0 +1,28 @@
...
+- Changes in `AlterTableRequest`:
+    - It now uses `table_name` instead of `Table`
+    - We have introduced a new field, `SchemaDiff`, which supports adding new columns, changing column type, and updating primary keys.
```
Comment (ediril) [inline CHANGELOG.md:25]:
```suggestion
    - Introduced a new field, `SchemaDiff`, which supports adding new columns, changing column type, and updating primary keys.  
```

File: CHANGELOG.md [hunk 36]
```diff
@@ -0,0 +1,28 @@
...
+- A new rpc method `Capabilities` is introduced. This brings the following functionalities:
+    - Adjust column types: Supports optionally adjusting Partner data type for each Fivetran type.
+    - Define the maximal value supported for the column types.
+    - We have added support for providing parquet batch files. Now, partner can choose the file type to create batch files.
+    - We have introduced a new field, `supports_history_mode`, which indicates whether the destination supports history mode.
+- Changes in `AlterTableRequest`:
+    - It now uses `table_name` instead of `Table`
+    - We have introduced a new field, `SchemaDiff`, which supports adding new columns, changing column type, and updating primary keys.  
+- Changes in WriteBatchRequest:
+  - We have added support for parquet batch files configuration details.
```
Comment (ediril) [inline CHANGELOG.md:27]:
```suggestion
  - Added support for parquet batch files configuration details.
```

File: CHANGELOG.md [hunk 37]
```diff
@@ -0,0 +1,28 @@
...
+  - We have introduced new field `HistoryMode`, which indicates that history mode is enabled for the table.
```
Comment (ediril) [inline CHANGELOG.md:28]:
```suggestion
  - Introduced a new field `HistoryMode`, which indicates that history mode is enabled for the table.
```

**Review by ediril — CHANGES_REQUESTED** [review]

**Comment by fivetran-abdulsalam** [comment]
@ediril, As per @5tran-alexil  suggestions I am changes from `We have introduced` to `Introduced` and from `We have added` to `Added`. Please check this comment: https://github.com/fivetran/fivetran_sdk/pull/44#discussion_r1644454002

File: CHANGELOG.md (hunk 19 above)
Comment (ediril) [reply CHANGELOG.md]: This is not how changelogs are written from what I've seen on the internet.. no need to use present perfect tense, past tense is sufficient. Also no need to specify "we".

File: CHANGELOG.md (hunk 32 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: Incorporated

File: CHANGELOG.md (hunk 31 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: Incorporated

File: CHANGELOG.md (hunk 25 above)
Comment (fivetran-abdulsalam) [reply CHANGELOG.md]: Incorporated

**Review by 5tran-alexil — APPROVED** [review]
LGTM

**Comment by fivetran-abdulsalam** [comment]
@ediril Could you please re-review this PR

**Comment by 5tran-alexil** [comment]
@fivetran-abdulsalam Is there something that blocks from merging this changelog?

**Comment by fivetran-abdulsalam** [comment]
Team has decided not to go with the changes and may go few changes in iteration. This PR is not needed now, so closing it

**Pull request closed without merging**